src/main/java/uk/gov/hmcts/divorce/common/event/CreateTestCase.java

```diff
@@ -0,0 +1,148 @@
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        var roles = new ArrayList<UserRole>();
+        var env = getenv().getOrDefault("S2S_URL_BASE", "aat");
+
+        if (env.contains(ENVIRONMENT_AAT)) {
+            roles.add(SOLICITOR);
+            roles.add(CASE_WORKER);
+            roles.add(SUPER_USER);
+        }
+
+        new PageBuilder(configBuilder
+            .event(TEST_CREATE)
+            .initialState(Draft)
+            .name("Create test case")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted)
+            .grant(CREATE_READ_UPDATE, roles.toArray(UserRole[]::new))
+            .grant(READ, SUPER_USER, CASE_WORKER, LEGAL_ADVISOR, SOLICITOR, CITIZEN))
+            .page("Create test case")
+            .mandatory(CaseData::getApplicationType)
+            .complex(CaseData::getApplicant1)
+                .mandatoryWithLabel(Applicant::getSolicitorRepresented, "Is applicant 1 represented")
+                .done()
+            .complex(CaseData::getApplicant2)
+                .mandatoryWithLabel(Applicant::getSolicitorRepresented, "Is applicant 2 represented")
+                .done()
+            .complex(CaseData::getCaseInvite)
+                .label("userIdLabel", "<pre>Use ./bin/get-user-id-by-email.sh [email] to get an ID"
+                    + ".\n\nTEST_SOLICITOR@mailinator.com is 93b108b7-4b26-41bf-ae8f-6e356efb11b3 in AAT.\n</pre>")
+                .mandatoryWithLabel(CaseInvite::getApplicant2UserId, "Applicant 2 user ID")
+                .done()
+            .complex(CaseData::getApplication)
+                .mandatoryWithLabel(Application::getStateToTransitionApplicationTo, "Case state")
+            .done();
+    }
+
+    @SneakyThrows
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        var file = details.getData().getApplicationType().isSole() ? SOLE_APPLICATION : JOINT_APPLICATION;
+        var resourceLoader = new DefaultResourceLoader();
+        var json = IOUtils.toString(resourceLoader.getResource(file).getInputStream(), Charset.defaultCharset());
+        var fixture = objectMapper.readValue(json, CaseData.class);
+
+        fixture.getApplicant1().setSolicitorRepresented(details.getData().getApplicant1().getSolicitorRepresented());
+        fixture.getApplicant2().setSolicitorRepresented(details.getData().getApplicant2().getSolicitorRepresented());
+        fixture.getCaseInvite().setApplicant2UserId(details.getData().getCaseInvite().getApplicant2UserId());
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(fixture)
+            .state(details.getData().getApplication().getStateToTransitionApplicationTo())
```

Comment:
we are setting the state here so we can remove `.initialState(Draft)` ?

Reply:
I think I need that to mark it as a creation event in CCD